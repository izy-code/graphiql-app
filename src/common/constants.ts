export const MIN_PASSWORD_LENGTH = 8;
export const EMAIL_REGEX = /^([A-Z0-9_+-]+\.?)*[A-Z0-9_+-]@([A-Z0-9][A-Z0-9-]*\.)+[A-Z]{2,}$/i;
export const LOWERCASE_LETTER_REGEX = /\p{Ll}/u;
export const UPPERCASE_LETTER_REGEX = /\p{Lu}/u;
export const SPECIAL_CHARACTER_REGEX = /[-+:|/\\%*#@$!?^&]/;
export const NUMBER_REGEX = /\d/;

export const NO_ENDPOINT = 'no-endpoint-provided';
export const STORE_RESET = 'store-reset-action';
