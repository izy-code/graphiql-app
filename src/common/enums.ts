export enum LocalStorageKeys {
  URLS_RSS_REQUEST = 'REQUEST_LIST',
}

export enum NonProtectedPaths {
  SIGN_UP = '/sign-up',
  SIGN_IN = '/sign-in',
}

export enum ProtectedPaths {
  REST = '/GET',
  GRAPHIQL = '/graphiql',
  HISTORY = '/history',
}
