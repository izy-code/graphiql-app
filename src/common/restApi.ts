export interface RestResponseData {
  status?: string;
  data?: object;
  errorMessage?: string;
}

export async function getResponse(
  method: string,
  endpoint: string,
  headers: Record<string, string> = {},
  body?: string,
): Promise<RestResponseData> {
  try {
    if (!endpoint) {
      return { errorMessage: 'restApi.errors.endpoint' };
    }

    const options: RequestInit = {
      method,
      headers,
      cache: 'no-store',
    };

    if (method !== 'GET' && body) {
      options.body = JSON.stringify(JSON.parse(body));
    }
    const response = await fetch(endpoint, options);
    const responseBody = (await response.json()) as { data?: object; errors?: object };

    if ('errors' in responseBody && Array.isArray(responseBody.errors) && responseBody.errors.length > 0) {
      return { errorMessage: 'restApi.errors.body' };
    }

    return { status: response.status.toString(), data: responseBody };
  } catch {
    return { errorMessage: 'restApi.errors.unknown', status: 'restApi.errors.status' };
  }
}
