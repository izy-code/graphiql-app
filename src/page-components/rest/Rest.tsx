'use client';

import { type ReactNode } from 'react';

import { AuthRoute } from '@/components/auth-route/AuthRoute';

import styles from './Rest.module.scss';

function Rest(): ReactNode {
  return (
    <div className={styles.page}>
      <h2>REST page</h2>
    </div>
  );
}

export default AuthRoute(Rest);
