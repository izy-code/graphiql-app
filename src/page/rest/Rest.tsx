'use client';

import { type ReactNode } from 'react';

import { AuthRoute } from '@/components/auth-route/AuthRoute';

import styles from './styles.module.scss';

function Rest(): ReactNode {
  return (
    <div className={styles.page}>
      <h2>REST page</h2>
    </div>
  );
}

export default AuthRoute(Rest);
