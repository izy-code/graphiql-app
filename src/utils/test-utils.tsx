import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { type PropsWithChildren, type ReactElement, useRef } from 'react';
import { Provider } from 'react-redux';

import type { AppStore, RootState } from '@/store/store';
import { setupStore } from '@/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProvidersAndUser(
  ui: ReactElement,
  { preloadedState = {}, store = setupStore(preloadedState), ...renderOptions }: ExtendedRenderOptions = {},
): { store: AppStore; user: ReturnType<typeof userEvent.setup>; container: HTMLElement } {
  function Wrapper({ children }: PropsWithChildren): ReactElement {
    const storeRef = useRef<AppStore>();

    if (!storeRef.current) {
      storeRef.current = setupStore(preloadedState);
    }

    return <Provider store={storeRef.current}>{children}</Provider>;
  }

  return { store, user: userEvent.setup(), ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderWithUserSetup(ui: ReactElement): {
  user: ReturnType<typeof userEvent.setup>;
  container: HTMLElement;
} {
  return {
    user: userEvent.setup(),
    ...render(ui),
  };
}
