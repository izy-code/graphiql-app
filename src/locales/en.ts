import { MIN_PASSWORD_LENGTH } from '@/common/constants';

export default {
  'header.links.sign-in': 'Sign in',
  'header.links.sign-up': 'Sign up',
  'header.button.sign-out': 'Sign out',

  'main.welcome': 'Welcome, please sign in.',
  'main.welcome.user': 'Welcome, {name}!',
  'main.title': 'REST/GraphQL client',
  'main.links.rest': 'REST client',
  'main.links.graphql': 'GraphQL client',
  'main.links.history': 'History',
  'main.links.sign-in': 'Sign in',
  'main.links.sign-up': 'Sign up',
  'main.project': 'Project information',
  'main.project-info':
    'Developed as the final assignment for a React course, this project showcases the coordinated effort of three dedicated participants.\nCompleted within a span of four weeks, the application is a result of the team’s hard work and collaboration.\nThe application supports both RESTful and GraphQL APIs, featuring Firebase authentication, request history tracking, and language switching.\nUsers can configure and send requests, view responses.',
  'main.team.name-1': 'Ivan Zhirebny',
  'main.team.role-1': 'Frontend-developer, Team leader',
  'main.team.contributions-1': 'Dev environment setup, Authentication pages, Firebase, GraphQL',
  'main.team.bio-1':
    'Graduated from the Bauman Moscow State Technical University, completed the Frontend Developer program at the HTML Academy',
  'main.team.name-2': 'Vadim Kolymbet',
  'main.team.role-2': 'Frontend-developer',
  'main.team.contributions-2': 'Routing, Internationalization, Testing',
  'main.team.bio-2':
    'Graduated from the Saratov State University, studied at the Faculty of Computer Science and Information Technologies and in the Department of Computer Security and Cryptography Theory. Graduated from layout courses at Frontendblok. Worked as a T-SQL developer',
  'main.team.name-3': 'Aleksandra Bodnar',
  'main.team.role-3': 'Frontend-developer, Scrum-master',
  'main.team.contributions-3': 'Design, Jira, MUI, REST, History page, Welcome page',
  'main.team.bio-3':
    'Graduated from the Immanuel Kant Baltic Federal University, studied at the Faculty of Business Informatics. Graduated from 3WC courses with a degree in Frontend-development. Built PHP-driven features for 1C CRM',
  'main.rs-1':
    'RSSchool is an online educational community focused on web development and programming, offering well-structured courses for both beginners and experienced developers.',
  'main.rs-2':
    'Renowned for its high-quality education, RSSchool provides a comprehensive curriculum covering everything from basic HTML, CSS, and JavaScript to advanced technologies like React and Node.js. The school emphasizes project-based learning, enabling students to build real-world projects and develop impressive portfolios.',
  'main.rs-3':
    'RSSchool thrives on a vibrant community of volunteers, including seasoned developers and alumni, who mentor and support new learners.',

  'teamMember.role': 'Role',
  'teamMember.contributions': 'Contributions',
  'teamMember.bio': 'Short bio',
  'teamMember.techStack': 'Tech stack: ',

  'sign-in.loader': 'Signing in...',
  'sign-in.title': 'Sign in',
  'sign-in.email': 'Email',
  'sign-in.password': 'Password',
  'sign-in.submit': 'Sign in',

  'sign-up.loader': 'Signing up...',
  'sign-up.title': 'Sign up',
  'sign-up.name': 'Name',
  'sign-up.email': 'Email',
  'sign-up.password.title': 'Password',
  'sign-up.password': 'Enter password',
  'sign-up.password.confirm': 'Confirm password',
  'sign-up.submit': 'Sign up',

  'rest.request.completed': 'The request has been completed, check the response field',
  'rest.title': 'REST client',
  'rest.endpoint': 'Endpoint URL',
  'rest.params': 'Params',
  'rest.headers': 'Headers',
  'rest.variables': 'Variables',
  'rest.body.title': 'Body: ',
  'rest.body': 'Body',

  'restApi.errors.fetch': 'Failed to fetch the response',
  'restApi.errors.endpoint': 'No endpoint provided',
  'restApi.errors.body': 'Response body contains errors',
  'restApi.errors.unknown': 'Unknown error occurred while making the request',
  'restApi.errors.status': 'Fetch error',

  method: 'Method',

  'clientTable.hide': 'Hide',
  'clientTable.show': 'Show',
  'clientTable.key': 'Key',
  'clientTable.value': 'Value',
  'clientTable.row': '+ row',
  'clientTable.delete': 'Delete',

  'requestButton.send': 'Send Request',

  'response.title': 'Response',
  'response.status': 'Status:',
  'response.body': 'Body:',

  'graphql.errors.parse': 'Can’t parse request body in URL',
  'graphql.schema.completed': 'The request has been completed, you can open the schema',
  'graphql.request.completed': 'The request has been completed, check the response field',
  'graphql.title': 'GraphQL client',
  'graphql.endpoint': 'Endpoint URL',
  'graphql.schema': 'Schema URL',
  'graphql.request': 'Request',
  'graphql.schema.request': 'Schema request',
  'graphql.schema.hide': 'Hide schema',
  'graphql.schema.show': 'Show schema',
  'graphql.params': 'Params',
  'graphql.query.title': 'Query: ',
  'graphql.query': 'Query',
  'graphql.headers': 'Headers',
  'graphql.variables.title': 'Variables: ',
  'graphql.variables': 'Variables',

  'schema.title': 'Schema docs:',

  'graphQlApi.response-errors.endpoint': 'No endpoint or query provided',
  'graphQlApi.response-errors.variables': 'Variables field is not valid JSON',
  'graphQlApi.response-errors.data': 'Unknown response structure: no "data" field. Check endpoint and request params',
  'graphQlApi.response-errors.unknown': 'Unknown error occurred while making the request',
  'graphQlApi.response-errors.status': 'Fetch error',

  'graphQlApi.schema-errors.endpoint': 'No schema endpoint provided',
  'graphQlApi.schema-errors.body': 'Schema docs response body contains errors',
  'graphQlApi.schema-errors.data':
    'Unknown response structure: no "data" or "__schema" fields. Check endpoint and headers',
  'graphQlApi.schema-errors.fetch-status': 'Schema docs fetch failed with status code: {status}',
  'graphQlApi.schema-errors.unknown': 'Unknown error occurred while making the schema request',

  'history.title': 'History',
  'history.empty.title': 'You haven’t executed any requests',
  'history.empty': 'It’s empty here. Try: ',
  'history.links.rest': 'REST client',
  'history.links.graphql': 'GraphQL client',
  'history.subtitle': 'Requests:',

  'error.title': 'Oops!',
  'error.text': 'Sorry, an unexpected error has occurred.',
  'error.desc': 'Error message:',
  'error.recommendation': 'Please try to refresh the page.',
  'error.refresh': 'Refresh the page',

  '404.message': 'The page you requested was not found.',
  '404.back': 'Previous page',
  '404.main': 'Main page',

  'auth.loading': 'Loading Firebase...',
  'auth.expired': 'Your token has expired, please sign in again',
  'auth.redirect': 'Redirecting...',

  'firebase.sign-out.success': 'You have been signed out. History of requests are cleared!',
  'firebase.sign-in.success': 'Successfully signed in with {email}!',
  'firebase.sign-up.success': 'Successfully signed up {name}!',
  'firebase.errors.USER_DISABLED': 'User disabled',
  'firebase.errors.INVALID_LOGIN_CREDENTIALS': 'Invalid credentials',
  'firebase.errors.EMAIL_EXISTS': 'Email already in use',
  'firebase.errors.TOO_MANY_ATTEMPTS_TRY_LATER': 'Too many authentication requests',
  'firebase.errors.NETWORK_REQUEST_FAILED': 'Network request failed',

  'validation.email.required': 'Email is required',
  'validation.email.format': 'Email must have valid format',
  'validation.password.required': 'Password is required',
  'validation.password.number': 'Password must contain a number',
  'validation.password.uppercase': 'Password must contain an uppercase letter',
  'validation.password.lowercase': 'Password must contain a lowercase letter',
  'validation.password.special': 'Password must contain one of: -+/%*:#@\\$!?|^&',
  'validation.password.min': `Password must be at least ${MIN_PASSWORD_LENGTH} characters`,
  'validation.name.required': 'Name is required',
  'validation.name.capital': 'Name must start with a capital letter',
  'validation.confirm.required': 'Please confirm your password',
  'validation.confirm.match': 'Passwords do not match',

  'customTextarea.prettifyError': 'Failed to prettify code. Check {language} syntax',
  'customTextarea.show': 'Show',
  'customTextarea.hide': 'Hide',
  'customTextarea.plainText': 'Plain text',
  'customTextarea.format': 'Prettify',
} as const;
