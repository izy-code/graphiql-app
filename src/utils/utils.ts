import {
  LOWERCASE_LETTER_REGEX,
  MIN_PASSWORD_LENGTH,
  NUMBER_REGEX,
  SPECIAL_CHARACTER_REGEX,
  UPPERCASE_LETTER_REGEX,
} from '@/common/constants';
import type { TableRow } from '@/components/client-table/types.ts';
import en from '@/locales/en';
import ru from '@/locales/ru';

const regexArray = [NUMBER_REGEX, UPPERCASE_LETTER_REGEX, LOWERCASE_LETTER_REGEX, SPECIAL_CHARACTER_REGEX];
export const MAX_STRENGTH = regexArray.length + 1;
export const getPasswordStrength = (password: string): number => {
  let strength = regexArray.reduce((result, regex) => result + Number(regex.test(password)), 0);
  strength += password.length >= MIN_PASSWORD_LENGTH ? 1 : 0;

  return strength;
};

export const translateText = (key: string): string =>
  window.location.pathname.split('/').includes('en') ? en[key as never] : ru[key as never];

export const encodeBase64 = (str: string): string => Buffer.from(str, 'utf-8').toString('base64').replace(/=+$/, '');

export const decodeBase64 = (str: string): string => {
  const pad = str.length % 4 === 0 ? '' : '===='.slice(str.length % 4);
  return Buffer.from(str + pad, 'base64').toString('utf-8');
};

export const generateUniqueId = (): string => `${Date.now()}-${Math.random().toString(36).substr(2, 9)}`;

export const replaceVariables = (text: string | undefined, variables: TableRow[]): string => {
  const variableMap = Object.fromEntries(
    variables.filter(({ key, value }) => key.trim() && value.trim()).map(({ key, value }) => [key, value]),
  );

  let result = text || '';
  Object.entries(variableMap).forEach(([variableKey, variableValue]) => {
    const variablePlaceholder = `{{${variableKey}}}`;
    result = result.replace(new RegExp(variablePlaceholder, 'g'), variableValue);
  });

  return result.trim();
};

export const getEncodedHeaders = (headersParameter: TableRow[]): string => {
  if (!headersParameter || headersParameter.length === 0) {
    return '';
  }

  const encodedHeaders = headersParameter
    .filter(({ key, value }) => key.trim() && value.trim())
    .map(({ key, value }) => `${encodeURIComponent(key)}=${encodeURIComponent(value)}`);

  return `?${encodedHeaders.join('&')}`;
};

export const updateHistory = (
  locale: string,
  method: string,
  endpoint: string,
  body: string,
  headers: TableRow[],
  variables: TableRow[],
): void => {
  const encodedEndpoint = endpoint ? encodeBase64(endpoint) : encodeBase64(' ');
  const encodedBody = encodeBase64(replaceVariables(body, variables) || ' ');
  const encodedHeaders = getEncodedHeaders(headers);

  window.history.replaceState(null, '', `/${locale}/${method}/${encodedEndpoint}/${encodedBody}${encodedHeaders}`);
};
