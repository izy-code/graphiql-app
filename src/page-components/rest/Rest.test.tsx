// import { screen, waitFor } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import { type ReactNode } from 'react';

import { getResponse } from '@/common/restApi';
import en from '@/locales/en';
import { renderWithProvidersAndUser } from '@/utils/test-utils';

import Rest from './Rest';

const translateREST = vi.fn((arg: string) => en[`rest.${arg}` as never]);
const translateResponse = vi.fn((arg: string) => en[`response.${arg}` as never]);
const translateTable = vi.fn((arg: string) => en[`clientTable.${arg}` as never]);
const translateRequestButton = vi.fn((arg: string) => en[`requestButton.${arg}` as never]);
const translateCode = vi.fn((arg: string) => en[`customTextarea.${arg}` as never]);

vi.mock('@/hooks/useAuth', () => ({ useAuth: vi.fn((): boolean => true) }));

vi.mock('@/locales/client', async (importOriginal) => {
  const actual = await importOriginal<object>();

  return {
    ...actual,
    useScopedI18n: vi.fn((arg: string) => {
      switch (arg) {
        case 'rest':
          return translateREST;
        case 'response':
          return translateResponse;
        case 'clientTable':
          return translateTable;
        case 'requestButton':
          return translateRequestButton;
        case 'customTextarea':
          return translateCode;
        default:
          return {};
      }
    }),
    useI18n: vi.fn(() => vi.fn(() => 'Method')),
    useCurrentLocale: vi.fn(() => 'en'),
  };
});

vi.mock('@uiw/react-codemirror', () => ({
  __esModule: true,
  default: ({
    value,
    onChange,
    onBlur,
  }: {
    value: string;
    onChange: (value: string) => void;
    onBlur: () => void;
  }): ReactNode => (
    <input value={value} onChange={(e) => onChange(e.target.value)} onBlur={onBlur} data-testid="codemirror" />
  ),
  EditorView: {
    theme: (): Record<string, unknown> => ({}),
  },
}));

describe('Rest page', () => {
  /*   it('renders correctly', () => {
    const { container } = render(<Rest responseData={{}} />);

    expect(container).toMatchSnapshot();
  }); */

  it('renders correctly', async () => {
    await mockRouter.push('/en/GET');
    const responseData = await getResponse('GET', `https://rickandmortyapi.com/api/character/2`, {});
    const { container } = renderWithProvidersAndUser(<Rest responseData={responseData} />);

    // expect(screen.getByText('Morty Smith')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
