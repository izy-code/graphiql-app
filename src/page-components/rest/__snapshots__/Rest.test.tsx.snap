// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rest page > renders correctly 1`] = `
<div>
  <div
    class="_page_e6b978"
  >
    <div
      class="_container_e6b978"
    >
      <h1
        class="_title_e6b978"
      >
        REST client
      </h1>
      <div
        class="_items_dd1cd5"
      >
        <h2
          class="_sectionTitle_dd1cd5"
        >
          URL
        </h2>
        <div>
          <div
            class="MuiFormControl-root _selectContainer_34c1c4 css-1uyh00o-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard _selectLabel_34c1c4 css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              Method
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl _select_34c1c4 css-n70jm4-MuiInputBase-root-MuiInput-root-MuiSelect-root"
            >
              <div
                aria-controls=":r0:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                id="demo-simple-select"
                role="combobox"
                tabindex="0"
              >
                GET
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="GET"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root _input_71d148 css-1rihfkf-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-standard css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r1:"
            id=":r1:-label"
          >
            Endpoint URL
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl css-1eed5fa-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id=":r1:"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="_section_dd1cd5"
      >
        <h2
          class="_sectionTitle_dd1cd5"
        >
          Params
        </h2>
        <div
          class="_container_8cb11c"
        >
          <div
            class="_titleContainer_8cb11c"
          >
            <h4
              class="_subtitle_8cb11c"
            >
              Headers
              :
            </h4>
            <button
              class="_button_1c0f93 _button--primary_1c0f93 _hideBtn_8cb11c _hidden_8cb11c"
              type="button"
            >
              <span
                class="visually-hidden"
              >
                Show
              </span>
            </button>
          </div>
        </div>
        <div
          class="_container_8cb11c"
        >
          <div
            class="_titleContainer_8cb11c"
          >
            <h4
              class="_subtitle_8cb11c"
            >
              Variables
              :
            </h4>
            <button
              class="_button_1c0f93 _button--primary_1c0f93 _hideBtn_8cb11c _hidden_8cb11c"
              type="button"
            >
              <span
                class="visually-hidden"
              >
                Show
              </span>
            </button>
          </div>
        </div>
        <div
          class="_item_dd1cd5"
        >
          <div
            class="_wrapper_a5b21b"
          >
            <h4
              class="_title_a5b21b"
            >
              Body: 
            </h4>
            <div
              class="_controls_a5b21b"
            >
              <button
                class="_button_1c0f93 _button--tertiary_1c0f93 _prettifyBtn_a5b21b"
                title="Prettify"
                type="button"
              >
                <span
                  class="visually-hidden"
                >
                  Prettify
                </span>
              </button>
              <label
                class="MuiStack-root css-si5z26-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  JSON
                </p>
                <span
                  class="MuiSwitch-root MuiSwitch-sizeMedium css-1p4ld5t-MuiSwitch-root"
                >
                  <span
                    class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
                  >
                    <input
                      class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                      type="checkbox"
                    />
                    <span
                      class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                  />
                </span>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  Plain text
                </p>
              </label>
              <button
                class="_button_1c0f93 _button--primary_1c0f93 _hideBtn_a5b21b _hidden_a5b21b"
                type="button"
              >
                <span
                  class="visually-hidden"
                >
                  Show
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="_center_3e6b58"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary _button_3e6b58 css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Send Request
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="_section_22b2aa _response_22b2aa"
    >
      <h2
        class="_sectionTitle_22b2aa"
      >
        Response
      </h2>
      <div
        class="_oneLine_22b2aa"
      >
        <h4
          class="_status_22b2aa"
        >
          Status:
        </h4>
        -
      </div>
      <div
        class="_body_22b2aa"
      >
        <div
          class="_wrapper_a5b21b"
        >
          <h4
            class="_title_a5b21b"
          >
            Body:
          </h4>
          <div
            class="_controls_a5b21b"
          />
        </div>
        <input
          data-testid="codemirror"
          value=""
        />
      </div>
    </div>
  </div>
  ;
</div>
`;
